### PR TITLE
Radiant temperature function

### PR DESCRIPTION
**Pull request by Shoes01**
This function is meant to generalize the radiant energy felt at one location. This allows things to radiate heat, like monsters!

At the moment there is just one strength of aura. This could be made dependent on the monster's size, or could be some weak/strong flags. For testing purposes, the flaming eye has a hot aura, and the shadow monster has a cold aura.

This is content salvaged from the PR #4975!

To be noted that the "warming limbs by nearby fire" was sort of left in the update_bodytemp() function, so code is a little duplicated... and also, blisters continue to be handled but I forget if they do anything anymore...

edit: as far as testing goes, I made a few fires and spawned a few monsters. Hallucinations don't have auras. I suppose I should have walked on the fire to ensure no divisions by zero :3

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -9653,8 +9735,8 @@ point game::look_around(WINDOW *w_info, const point pairCoordsFirst)
                         double f = double(range)/dist_f;
                         int tdx = lx - pos.x;
                         int tdy = ly - pos.y;
-                        tdx = std::round(f*tdx);
-                        tdy = std::round(f*tdy);
+                        tdx = round(f*tdx);
+                        tdy = round(f*tdy);
```
Comment (Shoes01) [inline src/game.cpp]: Oops, how did that get in there? Don't know what this is about.

File: src/game.cpp [hunk 2]
```diff
@@ -2281,6 +2281,88 @@ int game::get_temperature()
...
+        
+            if (tile_field.findField(fd_fire)) {
+                switch (tile_field.findField(fd_fire)->getFieldDensity()) {
+                    case 1:
+                        tile_radiant_temperature += fire_temperature_level[0]; break;
+                    case 2:
+                        tile_radiant_temperature += fire_temperature_level[0]; break;
+                    case 3:
+                        tile_radiant_temperature += fire_temperature_level[0]; break;
+                    case 4:
```
Comment (BevapDin) [inline src/game.cpp]: Field density can only go to 3. Is it intended that every case block is exactly the same (same index in the `fire_temperature_level` array)?

Comment (Shoes01) [inline src/game.cpp]: Good catch. Copy pasting error, and only testing a small fire error ;p

**Comment by Shoes01** [comment]
I had to rewrite some history while in a rush, I hope I didn't fudge anything up :x I would suggest not testing this until further notice.

**Comment by Shoes01** [comment]
I rewrote the history again. I had time to test it a little too, so fire seems to be as warm as one would think. Standing 2 tiles away from a burning building was intensely hot.

So this PR works as advertised now. Anything else anyone can think of that needs to radiate heat? Or other monsters that should have hot/cold auras?

**Comment by DavidKeaton** [comment]
I wanna look at the diff, upload the commit! :-)

**Comment by Barhandar** [comment]
>Anything else anyone can think of that needs to radiate heat?

Burning charcoal kilns, both item and furniture. Torches. Vehicle engines and generators.

In extreme case example, recipe radiant heat: while working the recipe's long action character has the radiant heat bumped by that amount. For forging (gets very hot), welding (same) and anything physically intensive.

**Comment by Shoes01** [comment]
I think because I rewrote history the commits are coming out the same. I didn't really add new code since the last update, just fixed an error or two? Honestly I don't quite remember :x

File: data/json/monsters.json [hunk 3]
```diff
@@ -4493,7 +4493,7 @@
     "death_function":"NORMAL",
     "special_attacks":[["STARE", 12]],
     "description":"An eyeball the size of an easy chair, covered in rolling blue flames.  It floats through the air, spinning slowly to draw its gaze in every direction.",
-    "flags":["SEES", "WARM", "FLIES", "FIREY", "NO_BREATHE", "NOHEAD", "FAT"]
+    "flags":["SEES", "WARM", "FLIES", "FIREY", "NO_BREATHE", "NOHEAD", "FAT", "HOTAURA"]
```
Comment (KA101) [inline data/json/monsters.json]: We should *probably* think of some other critter to receive the HOTAURA.  As is, the flaming eye's comparative vulnerability to melee is the only weak point it's got.

Comment (Shoes01) [inline data/json/monsters.json]: Noted. I will likely add it to the "charred nightmare" creature, and maybe create weak version of the auras to use on this creature and maybe others. Weak enough for a player to be like "damn, they thought of e'rythin'!"

**Comment by KA101** [comment]
Shoes, any updates?  I'd like to get this workable and in before it starts conflicting.

**Comment by KA101** [comment]
...I'm thinking new critters to receive the HOTAURA and a minor rework of rifts/hellmouths to accommodate 'em.  Any opinions?

**Comment by Shoes01** [comment]
Sorry I've been distracted by a different project. I will try to find time this weekend to put the finishing touches on this and get it in. I will probably not do the things Barhandar mentioned simply because the effect would be so tiny.

**Comment by KA101** [comment]
Critters I was thinking of would be some Nether fire-bats or somesuch, change the hellmouth's downstair to a portal, and have the bats spawn at the hellmouth.

**Comment by OzoneH3** [comment]
No progress for a long time, closing.

**Pull request closed without merging**